pkg/client/client.go

```diff
@@ -974,7 +974,21 @@ func (c *Client) WaitForDaemonSetRollout(ds *appsv1.DaemonSet) error {
 				d.Status.DesiredNumberScheduled, d.Status.UpdatedNumberScheduled)
 			return false, nil
 		}
-		if d.Status.NumberUnavailable != 0 {
+
+		var nodeNotReadyCount int32
+		nodeList, err := c.kclient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		nodes := nodeList.Items
+		for _, node := range nodes {
```

Comment:
(nit) saving one line

```suggestion
		for _, node := range nodeList.Items {
```

Reply:
Done